librz/util/str_trim.c

```diff
@@ -81,18 +81,23 @@ RZ_API char *rz_str_trim_dup(const char *str) {
 	return a;
 }
 
-// Returns a pointer to the first non-whitespace character of str.
-// TODO: Find a better name: to rz_str_trim_head_ro(), rz_str_skip_head or so
-RZ_API const char *rz_str_trim_head_ro(const char *str) {
+/* \brief Returns a pointer to the first non-whitespace character of \p str
+ *
+ * It considers space, TAB, and newline characters as the whitespace
+ */
+RZ_API const char *rz_str_trim_head_ro(RZ_NONNULL const char *str) {
```

Comment:
This should really be named `rz_str_first_non_whitespace`. i will open a PR after this is merged